### DISCUSSION TITLE
CSP on npm & suitability for use with components

### DISCUSSION THREAD

**Discussion by MikeHarrison** [alpinejs/alpine#2650] [1. Help]
Hi,

I have a couple of questions:

1. I can't find the CSP build on npm, is that because this version of Alpine is not complete? This project needs to have a CSP so I need to use this build
2. Are the new Alpine Components suitable for use with the CSP build (assuming it is ok for production use)? I understand I will need to adjust the syntax, but is the functonality compatible?

Thanks for any help!

**Comment by browner12** [comment]
The CSP build has not been deployed to NPM, so the instructions are not entirely accurate.

The only way to currently get the CSP build is clone the repo, run the build, and copy/paste the resulting ESM or CJS file.

Fair warning, I've found the CSP build to be basically unusable. Even something as simple as an x-show does not appear to be working.
